satellite/satellitedb/dbx/satellitedb.dbx

```diff
@@ -619,4 +619,26 @@ update offer (
 	where offer.expires_at >= offer.created_at
 )
 
-delete offer ( where offer.id = ? )
\ No newline at end of file
+delete offer ( where offer.id = ? )
+
+//--- user credit table ---//
+
+model user_credit (
+	key		user_id
+
+	field	user_id user.id	cascade
+	field	offer_id offer.id	cascade
+	field	referred_by user.id	cascade
```

Comment:
Same idea as above. When a user is deleted, we want the referred user still maintain the credits

Reply:
The documentation says the relation kind I should use for your suggestion is `restrict` so I will replace all instances of `cascade` with restrict.

From the dbx docs:

> Foreign Key Relation Kinds

A foreign key relation can be any of these

setnull: when the related row goes away, set this field to null. the field must be nullable.
cascade: when the related row goes away, delete this row.
restrict: do not allow the related row to go away.